pkg/devices/coils.go

```diff
@@ -161,13 +147,21 @@ func bulkReadCoils(managers []*ModbusDeviceManager) ([]*sdk.ReadContext, error)
 		}
 	}
 
-	if len(managersInErr) == len(managers) {
-		for _, manager := range managersInErr {
-			log.WithFields(log.Fields{
-				"host": manager.Host,
-				"port": manager.Port,
-			}).Error("[modbus] failed to read coils from host")
+	for _, manager := range managersInErr {
```

Comment:
This looks like a single function with calls in three places.

Reply:
good point, will update